src/txHandler/Bulker.TxHandler.ts

```diff
@@ -1,15 +1,194 @@
+
+import TransactionType = Bulker.TransactionType;
+
+export default class BulkerTxHandler
+  extends NotifierManager
+  implements IBatchTxHandler
+{
+  _operations: TxOperation[] = [];
+  _bulkerTransactions: Bulker.Transactions[] = [];
+  _value = constants.Zero;
+
+  public readonly operations$ = new Subject<TxOperation[]>();
+
+  protected set operations(ops: TxOperation[]) {
+    this._operations = ops;
+    this.operations$.next(ops);
+  }
+
+  public getOperations(): TxOperation[] {
+    return deepCopy(this._operations);
+  }
+  public getValues(): BigNumber {
+    return BigNumber.from(this._value);
+  }
+
+  public getBulkerTransactions(): Bulker.Transactions[] {
+    return deepCopy(this._bulkerTransactions);
+  }
+
+  constructor(protected _dataHolder: MorphoAaveV3DataHolder) {
+    super();
+  }
+
+  addOperation(operation: TxOperation): void {
+    switch (operation.type) {
+      case TransactionType.supply:
+      case TransactionType.supplyCollateral:
+        const { batch, value } = this.validateSupply(
+          operation.underlyingAddress,
+          operation.amount
+        );
+        this._value = this._value.add(value);
+        this._bulkerTransactions = [...this._bulkerTransactions, ...batch];
+        this.operations = [...this._operations, operation];
+        break;
+      default:
+        throw Error("Not implemented");
+    }
+  }
+
+  executeBatch(options: TransactionOptions | undefined): Promise<any> {
+    return Promise.resolve(undefined);
+  }
+
+  removeOperation(index: number): void {
+    if (index < 0 || index >= this._operations.length) {
+      throw new Error("Invalid index");
+    }
+    const newOperations = [...this._operations.filter((_, i) => i !== index)];
+
+    //  TODO:  validate
+    this.operations = newOperations;
+  }
+
+  validateSupply(
+    underlyingAddress: string,
+    amount: BigNumber
+  ): { value: BigNumber; batch: Bulker.Transactions[] } {
+    if (amount.isZero()) throw Error("Amount is zero");
+    const batch: Bulker.Transactions[] = [];
+    let deferSkimWsteth = false;
+    let value = constants.Zero;
+
+    const userMarketsData = this._dataHolder.getUserMarketsData();
+    const userData = this._dataHolder.getUserData();
+    if (!userData || !userMarketsData)
+      throw Error("User data or user markets data is undefined");
+    let toTransfer = amount;
+    if (getAddress(underlyingAddress) === addresses.wsteth) {
+      const wstethMissing = maxBN(
+        amount.sub(userMarketsData[underlyingAddress]!.walletBalance),
+        constants.Zero
+      );
+      if (wstethMissing.gt(0)) {
+        // we need to wrap some stETH
+        // To be sure that  , we add 1e8 to the amount wrapped
+        const WRAP_BUFFER = BigNumber.from(1e8);
+        const amountToWrap = WadRayMath.wadMul(
+          wstethMissing.add(WRAP_BUFFER),
+          userData.stEthData.stethPerWsteth
+        );
+
+        const stEthAmountToWrapWithBuffer = amountToWrap.add(WRAP_BUFFER);
```

Comment:
You already added the buffer at line 105

Reply:
Ok that explains why Ive a 2e9 value in my tests lol

Reply:
That's what I thought as well haha